lib/msal-node/docs/managed-identity.md

```diff
@@ -0,0 +1,139 @@
+
+[acquireToken](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/msi_feature_branch/lib/msal-node/src/client/ManagedIdentityApplication.ts#L122) is called with the resource to acquire a token for, such as `https://management.azure.com`.
+
+```typescript
+// optional
+const config: ManagedIdentityConfiguration = {
+    system: {
+        loggerOptions: {
+            logLevel: LogLevel.Verbose,
+            loggerCallback: (_level, message, _containsPii) => {
+                console.log(message);
+            },
+            piiLoggingEnabled: true,
+        },
+    },
+};
+
+const systemAssignedManagedIdentityApplication = new ManagedIdentityApplication(
+    config
+);
+
+const managedIdentityRequest: ManagedIdentityRequestParams = {
+    resource: "https://management.azure.com",
+};
+
+try {
+    const response =
+        await systemAssignedManagedIdentityApplication.acquireToken(
+            managedIdentityRequest
+        );
+    console.log(response);
+} catch (error) {
+    console.log(error);
+    throw error;
+}
+```
+
+### User-assigned managed identities
+
+For user-assigned managed identities, the developer needs to pass either the client ID, full resource identifier, or the object ID of the managed identity when creating ManagedIdentityApplication.
+
+Like in the case for system-assigned managed identities, ` acquireToken` is called with the resource to acquire a token for, such as `https://management.azure.com`.
+
+```typescript
+const managedIdentityIdParams: ManagedIdentityIdParams = {
+    userAssignedClientId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+};
+
+const config: ManagedIdentityConfiguration = {
+    managedIdentityIdParams,
+    // optional
+    system: {
+        loggerOptions: {
+            logLevel: LogLevel.Verbose,
+            loggerCallback: (_level, message, _containsPii) => {
+                console.log(message);
+            },
+            piiLoggingEnabled: true,
+        },
+    },
+};
+
+const userAssignedClientIdManagedIdentityApplication =
+    new ManagedIdentityApplication(config);
+
+const managedIdentityRequest: ManagedIdentityRequestParams = {
+    resource: "https://management.azure.com",
+};
+
+try {
+    const response =
+        await userAssignedClientIdManagedIdentityApplication.acquireToken(
+            managedIdentityRequest
+        );
+    console.log(response);
+} catch (error) {
+    console.log(error);
+    throw error;
+}
+```
+
+## Caching
+
+By default, MSALJS supports in-memory caching. Managed identity in MSALJS does not support cache extensibility because of security concerns when using distributed cache. Since a token acquired for managed identity belongs to an Azure resource, using a distributed cache might expose it to the other Azure resources sharing the cache.
```

Comment:
```suggestion
MSALJS caches tokens from managed identity in memory. There is no eviction, but memory is not a concern because there can only exist a limited number of managed identities defined. Cache extensibility is not supported in this scenario because tokens should not be shared between machines.
```